site/src/pages/tutorials/grpc/blog/add-service-to-server.mdx

```diff
@@ -0,0 +1,187 @@
+---
+menuTitle: "Run blog service on server"
+order: 3
+type: step
+tags:
+  - docservice
+  - annotated-services
+level: basic
+---
+
+# Running blog service on a server
+
+In this step, we'll create a server instance and to it add the service we implemented in [Step 2. Implementing a gRPC service](/tutorials/grpc/blog/implement-grpc-service). Also, we'll run Armeria's [Documentation service](/docs/server-docservice) along with our blog service to test our service.
+
+<TutorialSteps current={3} />
+
+## What you need
+
+You need to have the blog service [defined](/tutorials/grpc/blog/define-service) and [implemented](/tutorials/grpc/blog/implement-grpc-service).
+
+## 1. Create a Main class
+
+Create a `Main` class. You can see the full version of the file [here](https://github.com/line/armeria-examples/blob/master/tutorials/rest-api-annotated-service/src/main/java/example/armeria/server/blog/Main.java).
+
+```java filename=Main.java
+package example.armeria.server.blog;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Main {
+  private static final Logger logger = LoggerFactory.getLogger(Main.class);
+
+  public static void main(String[] args) throws Exception {
+      final Server server = newServer(8080);
+
+      Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+          server.stop().join();
+          logger.info("Server has been stopped.");
+      }));
```

Comment:
I find this pattern fairly often. Should we add a shortcut for this to `Server` and `ClientFactory`? e.g.

```
server.closeOnShutDown()
clientFactory.closeOnShutdown()
```

Reply:
Nice idea! Let me file an issue for it.